redash/settings.py

```diff
@@ -73,6 +73,8 @@ def all_settings():
 CELERY_BROKER = os.environ.get("REDASH_CELERY_BROKER", REDIS_URL)
 CELERY_BACKEND = os.environ.get("REDASH_CELERY_BACKEND", CELERY_BROKER)
 CELERY_TASK_RESULT_EXPIRES = int(os.environ.get('REDASH_CELERY_TASK_RESULT_EXPIRES', 3600))
+CELERYD_PREFETCH_MULTIPLIER = int(os.environ.get('REDASH_CELERYD_PREFETCH_MULTIPLIER', 4))
+CELERY_ACKS_LATE = parse_boolean(os.environ.get('REDASH_CELERY_ACKS_LATE', "false"))
```

Comment:
`ACKS_LATE` is a dangerous setting to add, as anyone who doesn't fully understand how Celery works can easily shoot himself in the foot.

If I understand it correctly, when Celery acknowledges tasks only after they finish (ACKS_LATE = True), then if the task takes > `visibility_timeout` (default = 1 hour) it will run the task again. Meaning that any query that takes longer than `visibility_timeout` will be executed multiple times.

The prefetch multiplier is only relevant in cases where there are more than one Celery workers for the same queue (not the same thing as one worker with a process pool). It only applies to _unacknowledged_ tasks, so in the case of ACKS_LATE=False, any task the worker fetches will be immediately acknowledged...

Relevant documentation:
http://docs.celeryproject.org/en/3.1/glossary.html
http://docs.celeryproject.org/en/3.1/getting-started/brokers/redis.html

This is another example of how Celery is just too clever for us and we might benefit from using a simpler library like [rq](http://python-rq.org/)...

Reply:
Thanks @jeffwidman. Actually I'm not sure about "Celery's best backend is redis" -- at some point they even thought to deprecate the Redis support. I think the most compatible backend is RabbitMQ.

My main issue w/ Celery is visibility into what's going on (# of currently running jobs, active workers, etc). It has some APIs to get this information, but it's not stable from my experience if the workers are overloaded and can't respond to the messages.

I think going forward we will either have to better understand how to monitor Celery with Redis or switch to `rq` while implementing missing functionality.